### DISCUSSION TITLE
Doubt about Microsoft Community Promise

### DISCUSSION THREAD

**Discussion by federico-r-figueredo** [mono/mono#21541] [General]
I'd like to know if I fork this project to make my own version of the mono platform, would I still be covered by the Microsoft Community Promise asserted in this [file](https://github.com/dotnet/runtime/blob/main/PATENTS.TXT) too? Or that only applies to the original mono repo? If I make modifications to extend it that go beyond the ECMA standards or the non standardized parts of .NET Framework implemented in this project, would I loose that CP protection & be a (potential) target for a lawsuit by Microsoft? Would I be in the same risk if my fork is modified to a point in which it no longer complies w/ the ECMA 334 & ECMA 335? (e.g.: being almost a new programming language & runtime)

**Comment by akoeplinger** [comment]
Yes, we have the same Patent Promise in this repo here: https://github.com/mono/mono/blob/main/PATENTS.TXT

So go ahead and fork :)
